test/livebook/live_markdown/markdown_helpers_test.exs

```diff
@@ -243,6 +361,12 @@ defmodule Livebook.LiveMarkdown.MarkdownHelpersTest do
       assert markdown == reformat(markdown)
     end
 
+    test "surprise ordered list" do
+      markdown = "1986\\. What a great season."
+
+      assert markdown == reformat(markdown)
```

Comment:
This fails, again because the `\` is devoured by the first formatting pass.

Example of an "accidental list" taken from the Markdown docs.